python_modules/dagster/dagster/_core/storage/runs/migration.py

```diff
@@ -412,3 +414,33 @@ def add_backfill_job_name(run_storage: RunStorage, backfill_id: str, job_name: s
             )
             .where(BulkActionsTable.c.key == backfill_id)
         )
+
+
+def migrate_backfill_end_timestamp(storage: RunStorage, print_fn: Optional[PrintFn] = None) -> None:
+    """Utility method to add a backfill's end timestamp to the serialized backfill stored in the bulk actions table."""
+    if print_fn:
+        print_fn("Querying run storage.")
+
+    for backfill in chunked_backfill_iterator(storage, print_fn):
+        if backfill.status == BulkActionStatus.REQUESTED:
+            # we don't want to mutate a backfill that is still in progress. Additionally, it won't
+            # have an end timestamp until it moves to a terminal state
+            continue
+        if backfill.backfill_end_timestamp is None:
+            end_time = get_end_timestamp_for_backfill(storage, backfill)
+            updated_backfill = backfill.with_end_timestamp(end_time)
+            storage.update_backfill(updated_backfill)
+
+
+def get_end_timestamp_for_backfill(run_storage: RunStorage, backfill: PartitionBackfill) -> float:
+    filters = RunsFilter.for_backfill(backfill.backfill_id)
```

Comment:
maybe an assert in here that the backfill is in a terminal state?

in the off-chance that we add a new status but don't update this migration code.